### PR TITLE
feat: Integrate config service for NEST (#14)

### PR DESCRIPTION
**Pull request by Persists**
Introduce enhanced configuration validationg feature. This PR adds a new feature that improves the validation of environment variables for configuration parameters. It enhances the `validateConfig` function to ensure the correctness of the configuration data and sets default values where necessary.

This feature enhances the robustness and flexibility by introducing the config service.

### CODE REVIEW

File: apps/backend/src/app/app.controller.spec.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ describe('AppController', () => {
 
 	beforeAll(async () => {
 		app = await Test.createTestingModule({
+
```
Comment (KonsumGandalf) [inline apps/backend/src/app/app.controller.spec.ts:11]:
```suggestion
```

Weg mit der Zeile

File: apps/backend/src/app/config/classes/environment.class.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	@IsDefined()
+	@IsString()
+	@MinLength(1)
+	POSTGRES_USER: string;
+
+	@IsDefined()
+	@IsString()
+	@MinLength(1)
+	POSTGRES_PASSWORD: string;
```
Comment (KonsumGandalf) [inline apps/backend/src/app/config/classes/environment.class.ts:43]: Some sort of docs would be good 🙏

File: apps/backend/src/main.ts [hunk 3]
```diff
@@ -4,15 +4,22 @@
...
 
 async function bootstrap() {
 	const app = await NestFactory.create(AppModule);
+
+	const configService = app.get(ConfigService<BackendConfig>);
+
 	const globalPrefix = 'api';
 	app.setGlobalPrefix(globalPrefix);
-	const port = process.env.PORT || 3000;
+	const port = configService.get('app', { infer: true }).port || 3333;
```
Comment (KonsumGandalf) [inline apps/backend/src/main.ts:22]: Please decide whether you are using port 3333 or 3000 as default, currently main.ts and validation.ts use different fallback values

File: docker-compose.yml [hunk 4]
```diff
@@ -39,6 +39,8 @@ services:
             - '3000:3000'
         networks:
             - backend_network
+        env_file:
+            - .env
```
Comment (KonsumGandalf) [inline docker-compose.yml:43]: What do you think about creating the .env file inside the app/backend dir? Then we could do it the same for the frontend .env. Additionally I would create a .postman.env, which contains just the information needed for the postgres docker image. => You would need to adjust configService as well, if you chose to do so. 
```ts

	imports: [
		ConfigModule.forRoot({
			isGlobal: true,
			envFilePath: ['.env'], // here
			validate: validateConfig,
		}),
```

File: libs/common/backend/models/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# common-backend-models
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `nx test common-backend-models` to execute the unit tests via [Jest](https://jestjs.io).
```
Comment (KonsumGandalf) [inline libs/common/backend/models/README.md:7]: Some docs would be good, I guess not all project members have used DDD yet. Therefore a description might be helpful for their understanding.

File: libs/common/backend/models/src/interfaces/app.interface.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+  /**
+   * The host the application will run on
+   * @type {string}
+   */
+  host: string;
+
+  /**
+   * The name of the application
+   * @type {string}
```
Comment (KonsumGandalf) [inline libs/common/backend/models/src/interfaces/app.interface.ts:19]: I like that you always anotate the type, but I guess the IDE does this automatically (just a remark not need for changing it)

File: package.json [hunk 7]
```diff
@@ -13,24 +13,33 @@
...
+    "class-validator": "^0.14.1",
+    "dotenv": "^16.4.5",
+    "lodash": "^4.17.21",
+    "postgres": "^3.4.4",
+    "prisma": "^5.13.0",
+    "reflect-metadata": "^0.1.13",
+    "rxjs": "~7.8.0",
+    "ts-enum-util": "^4.1.0",
+    "tslib": "^2.3.0",
+    "zone.js": "~0.14.3"
```
Comment (KonsumGandalf) [inline package.json:42]: Reformat!

File: libs/common/backend/models/tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  "include": [],
+  "references": [
+    {
+      "path": "./tsconfig.lib.json"
+    },
+    {
+      "path": "./tsconfig.spec.json"
+    }
+  ]
+}
```
Comment (KonsumGandalf) [inline libs/common/backend/models/tsconfig.json:22]: Reformat (2 Intent, we use 4)

**Review by KonsumGandalf — CHANGES_REQUESTED** [review]
I like the way you document most files! 😇
However please ensure next time, that you format the files before uploading them.

**Review by Sabo2k — APPROVED** [review]

File: apps/backend/src/app/app.controller.spec.ts (hunk 1 above)
Comment (Persists) [reply apps/backend/src/app/app.controller.spec.ts:11]: Done

File: apps/backend/src/main.ts (hunk 3 above)
Comment (Persists) [reply apps/backend/src/main.ts:22]: As i am already setting a default, i will remove this

File: libs/common/backend/models/tsconfig.json (hunk 8 above)
Comment (Persists) [reply libs/common/backend/models/tsconfig.json:22]: Thank you

File: docker-compose.yml (hunk 4 above)
Comment (Persists) [reply docker-compose.yml:43]: we could do that yes, i thought we would go for  one single .env, but i can refactor it

Comment (Persists) [reply docker-compose.yml]: Ok moved it to the backend Thanks for the tipp

File: libs/common/backend/models/README.md (hunk 5 above)
Comment (Persists) [reply libs/common/backend/models/README.md]: Did so :)

Comment (KonsumGandalf) [reply libs/common/backend/models/README.md]: Perfect

File: libs/common/backend/models/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export * from "./classes"
+export * from "./interfaces"
```
Comment (KonsumGandalf) [inline libs/common/backend/models/src/index.ts:2]: reformat

**Review by KonsumGandalf — CHANGES_REQUESTED** [review]
Except for the reformating I agree of it :)

File: libs/common/backend/models/src/index.ts (hunk 9 above)
Comment (Persists) [reply libs/common/backend/models/src/index.ts]: Done

**Review by KonsumGandalf — APPROVED** [review]

**Comment by sn0wcat** [comment]
@KonsumGandalf you already have `.prettierrc' config ,you just need to add a task which checks the formatting in the lint step. For example you need to run this in CI/CD pipeline `prettier --check .`

You can also use `husky` to disallow checking in of the code if it is not properly formatted

**Comment by sn0wcat** [comment]
@KonsumGandalf adding a task which automatically formats all source code `prettier --write .` is probably also a good idea.

**Pull request merged**